presto-main/src/main/java/io/prestosql/execution/CreateTableTask.java

```diff
@@ -170,8 +182,12 @@ else if (element instanceof LikeClause) {
                     inheritedProperties = likeTableMetadata.getMetadata().getProperties();
                 }
 
-                likeTableMetadata.getColumns().stream()
-                        .filter(column -> !column.isHidden())
+                accessControl.filterColumns(
```

Comment:
What is happening when all columns are filtered? Do you have test for that?

Reply:
No columns are added then. Is that not a desired outcome?

Reply:
That is the question. I think it is not a desired behavior. Notice wants to access some columns but he has no access to any, so result could be unexpected.

Also the more I think about it I think we should consider changing the approach. `LIKE existing_table_name` is like `SELECT *` so user wants to access all columns and in case we filter any of them we might need to raise an error.

@martint What are your thoughts about it? Is `CREATE TABLE table_name (LIKE existing_table_name)` a ANSI SQL defined behaviour?

Reply:
Yes, that is ANSI SQL behavior. This is what the spec says regarding permissions:

```
2) If a <like clause> is contained in a <table definition>, then the applicable privileges for A shall include SELECT privilege on the table identified in the <like clause>.
```

Reply:
> A shall include SELECT privilege on the table identified

That translates to having SELECT privilege on all columns.

So my suggestion here is to verify two things here:
 - check if user have SELECT privilege on all columns `on the table identified in the <like clause>`
 - if properties are included, then we also need to check if user `checkCanShowCreateTable` `on the table identified in the <like clause>` to prevent leakage of sensitive information like storage location

@martint do you agree?

Reply:
Yes, that makes sense.